### PR TITLE
[412] refactor(HXElement): refactor behavior of multiple custom elements

### PR DESCRIPTION
**Pull request by CITguy**
* Accordion
  * use new `replaceWith` util
* Disclosure
  * defer `$onConnect` to give target a chance to initialize
  * add private methods to abstract target event listener subscriptions
  * fix `target` getter logic to be aware if connected
* Tabset
  * defer `$onConnect` to let children initialize before parent
  * refactor to use new util methods
* Tooltip
  * refactor to use new `generateId` util method

JIRA: SURF-1515

### LGTM's
- [x] Dev LGTM
- [x] Zoom LGTM

### CODE REVIEW

**Comment by HelixUI** [comment]
Deploy preview for *helix-ui* processing.

Built with commit 268292bab9e7c49272a9d9b6db34ed184c8820bc

https://app.netlify.com/sites/helix-ui/deploys/5c34f1423c672b0008d76a30

File: src/helix-ui/utils/index.js [hunk 1]
```diff
@@ -55,6 +55,57 @@ export const KEYS = {
     Up: 38,
 };
 
+/**
+ * Defers execution of callback function until _next_ event loop.
+ *
+ * @param {Function} callback
+ */
```
Comment (100stacks) [inline src/helix-ui/utils/index.js]: add a comment `@return`

File: src/helix-ui/utils/index.js [hunk 2]
```diff
@@ -55,6 +55,57 @@ export const KEYS = {
...
+ * Generate a unique ID
+ *
+ * **Pseudo-random Algorithm**
+ * This functionality is pseudo-random, and you should not depend on 100%
+ * random values. Given a large enough dataset, this method has the
+ * potential to generate duplicate values.
+ *
+ * For the purposes of most applications, the dataset is small enough that
+ * the potential for duplicate values is almost 0, meaning that it's good
+ * enough for use.
```
Comment (100stacks) [inline src/helix-ui/utils/index.js]: if possible, add a link to the SO for this.

File: src/helix-ui/utils/index.js [hunk 3]
```diff
@@ -81,31 +132,42 @@ export function onScroll (evt) {
...
 }
+
+// Not everything needs to be part of the default export
+export default {
+    KEYS,
+    Position,
+    generateId,
+    onScroll,
+    preventKeyScroll,
+};
```
Comment (100stacks) [inline src/helix-ui/utils/index.js]: please check to see if we're exporting everything by default.

File: src/helix-ui/utils/index.js (hunk 2 above)
Comment (CITguy) [reply src/helix-ui/utils/index.js]: Couldn't find the original inspiration, but found a gist that is pretty darn close.

**Pull request merged**